@pvalsecc please test it. 
The fix is for reading from XML so there's no going from a value to another (other than the hard-coded arbitrary value of 100-100000). Setting min scale affects the max scale, so when setting both you want to set _min_ before _max_. I guess a further improvement in this PR would be to make every enforced clamping clearly logged with a WARNING.

I'm still working on an automated test, so different cases can be put in observation.
Right now I'm stuck with an error I cannot get my head around:
```
Generating moc source testqgsscalerangewidget.moc
[ 97%] Built target qgis_scalerangewidget_automoc
Scanning dependencies of target qgis_scalerangewidget
[ 97%] Building CXX object tests/src/gui/CMakeFiles/qgis_scalerangewidget.dir/testqgsscalerangewidget.cpp.o
In file included from /usr/src/qgis/qgis-2.18/tests/src/gui/testqgsscalerangewidget.cpp:87:0:
/usr/src/qgis/build/2.18/tests/src/gui/testqgsscalerangewidget.moc:18:1: error: expected constructor, destructor, or type conversion before ‘static’
 static const uint qt_meta_data_TestQgsScaleRangeWidget[] = {
 ^~~~~~
/usr/src/qgis/build/2.18/tests/src/gui/testqgsscalerangewidget.moc:72:7: error: ‘qt_meta_data_TestQgsScaleRangeWidget’ was not declared in this scope
       qt_meta_data_TestQgsScaleRangeWidget, &staticMetaObjectExtraData }
       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [tests/src/gui/CMakeFiles/qgis_scalerangewidget.dir/testqgsscalerangewidget.cpp.o] Error 1
make[1]: *** [tests/src/gui/CMakeFiles/qgis_scalerangewidget.dir/all] Error 2
```

The .moc file is generated so I dunno why it fails like that. Does it ring a bell for anyone ?

--- a/src/gui/qgsscalerangewidget.cpp
+++ b/src/gui/qgsscalerangewidget.cpp
@@ -132,8 +132,8 @@ double QgsScaleRangeWidget::maximumScaleDenom()
 
 void QgsScaleRangeWidget::setScaleRange( double min, double max )
 {
-  setMaximumScale( max );
   setMinimumScale( min );
+  setMaximumScale( max );
 }
 
 void QgsScaleRangeWidget::emitRangeChanged()

